mmcv/cnn/rfsearch/operator.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+from typing import Dict, Optional
+
+import numpy as np
+import torch
+import torch.nn as nn
+from torch import Tensor
+
+from mmcv.cnn.rfsearch.utils import expands_rate, get_single_padding
+from mmcv.runner import BaseModule
+from mmcv.utils.logging import get_logger
+
+logger = get_logger('Operators')
+
+
+class ConvRFSearchOp(BaseModule):
+    """Based class of ConvRFSearchOp.
+
+    Args:
+        op_layer (nn.Module): pytorch module, e,g, Conv2d
+        global_config (Dict): config dict.
+    """
+
+    def __init__(self, op_layer: nn.Module, global_config: Optional[Dict]):
+
+        super().__init__()
+        self.op_layer = op_layer
+        self.global_config = global_config
+
+    def normlize(self, weight: nn.Parameter) -> nn.Parameter:
+        """Normalize weights.
+
+        Args:
+            w (nn.Parameter): unnormed weights
+
+        Returns:
+            nn.Parameters: normed weights
+        """
+        abs_weight = torch.abs(weight)
+        norm_weight = abs_weight / torch.sum(abs_weight)
+        return norm_weight
+
+
+class Conv2dRFSearchOp(ConvRFSearchOp):
+    """Enable Conv2d with rf search ability.
+
+    Args:
+        op_layer (nn.Module): pytorch module, e,g, Conv2d
+        init_dilation (list, optional): init dilation rate. Defaults to None.
+        global_config (dict): config dict. Defaults to None.
+            By default this must includes:
+            - "init_alphas": The value for initializing weights of each branch.
+            - "num_branches": The controller of the size of
+                search space (the number of branches).
+            - "exp_rate": The controller of the sparsity of search space.
+            - "mmin": The minimum dilation rate.
+            - "mmax": The maximum dilation rate.
+            Extra keys may exist, but are used by RFSearchHook, e.g., "step",
+            "max_step", "search_interval", and "skip_layer".
+        num_branches (int, optional): Number of branches. Defaults to 3.
+        verbose (bool): Determines whether to print rf-next
+            related logging messages.
+            Defaults to True.
+    """
+
+    def __init__(self,
+                 op_layer: nn.Module,
+                 init_dilation: list = None,
+                 global_config: Optional[Dict] = None,
```

Comment:
```suggestion
                 global_config: dict,
                 init_dilation: list = None,
```

Reply:
`global_config` does not accept `None` so it would be better to remove the default value `None` from it.